umm-lib/src/cmr/umm/iso_smap/granule/orbit.clj

```diff
@@ -112,4 +168,39 @@
        (when end-direction
          (let [ed (subs orbit-str
                         end-direction)]
-           (convert-direction (str/trim (subs ed (inc (.indexOf ed ":")))))))))))
+           (convert-direction (str/trim (subs ed (inc (.indexOf ed ":"))))))))
+      "OrbitCalculatedSpatialDomains"
+      (g/->OrbitCalculatedSpatialDomain
```

Comment:
This is pretty awkward...  essentially, a bunch of logic is getting jammed into a constructor call, right? Also, positional args are harder to read and debug for future work. How about having a `let`  block where the args for the constructor are defined, and then use `(g/map->OrbitCalculatedSpatialDomain {:field1 field1 ...}`? You don't have to worry about nil values, since any values not supplied to a record are set to nil anyway ... so that will get rid of all the `when`s ...

Reply:
Yes it is!  Will rewrite this. Thanks.